### PR TITLE
Execute CiviRules upgrade sql statements only if CiviRules is installed

### PR DESCRIPTION
**Pull request by haraldballuch**

### CODE REVIEW

File: CRM/Emailapi/Upgrader.php [hunk 1]
```diff
@@ -52,10 +52,12 @@ public function upgrade_1002() {
    * update class name of the send e-mail action and add the send e-mail to related contact
    */
   public function upgrade_1003() {
-    CRM_Core_DAO::executeQuery("UPDATE civirule_action SET class_name = 'CRM_Emailapi_CivirulesAction_Send' WHERE `name` = 'emailapi_send'");
-    CRM_Core_DAO::executeQuery("INSERT INTO civirule_action (name, label, class_name, is_active)
-      VALUES('emailapi_send_relationship', 'Send E-mail to a related contact', 'CRM_Emailapi_CivirulesAction_SendToRelatedContact', 1);"
-    );
+    if(civicrm_api3('Extension', 'get', ['key' => 'civirules', 'status' => 'installed'])['count']){
```
Comment (jaapjansma) [inline CRM/Emailapi/Upgrader.php]: This code does not work. The civirules extension still has the key org.civicoop.civirules. And a query on Extension Get with status filter on Installed does return all extensions. 

Have a look at: https://lab.civicrm.org/partners/civicoop/cvvede/nl.cvvede.automation/blob/master/CRM/Automation/Upgrader.php#L24 for a working example.

**Review by jaapjansma — CHANGES_REQUESTED** [review]
The statement whether CiviRules is installed is not going to work.

File: CRM/Emailapi/Upgrader.php (hunk 1 above)
Comment (haraldballuch) [reply CRM/Emailapi/Upgrader.php]: Ok, sorry, I didn't test that but only copied the conditionals from function install() and upgrade_1002() in the same file to upgrade_1003(). 
But as I see now, after your comment this actually doesn't work. I'll come up with a different suggestion for all the mentioned functions.

**Comment by haraldballuch** [comment]
Using "civicrm_api3('Extension', 'get', ['full_name' => 'org.civicoop.civirules', 'status' => 'installed'])['count']" works and only returns the requested extension with the given key if it has the status "installed".

It seems that "key" and "full_name" both work in the same way.

**Review by jaapjansma — APPROVED** [review]

**Pull request merged**